api/src/main/java/org/apache/iceberg/Scan.java

```diff
@@ -93,6 +100,18 @@
    */
   T planWith(ExecutorService executorService);
 
+  /**
+   * Returns this scan's projection {@link Schema}.
+   * <p>
+   * If the projection schema was set directly using {@link #project(Schema)}, returns that schema.
+   * <p>
+   * If the projection schema was set by calling {@link #select(Collection)}, returns a projection
+   * schema that includes the selected data fields and any fields used in the filter expression.
+   *
+   * @return this scan's projection schema
+   */
+  Schema schema();
```

Comment:
This is moved from `TableScan` to base `Scan` interface as it is used by `ScanTestBase`.